client/src/components/PhotoGrid/PhotoGrid.jsx

```diff
@@ -35,14 +39,13 @@ function PhotoGrid(props) {
         img.src = src.imageUrl;
         img.onload = resolve();
         img.onerror = reject();
-        console.log(img)//!THIS CHECKS IF IT RAN
       })
     )))
   );
 
   useEffect(() => {
-    getPhotos(listingId);
-  }, []);
+    cacheImages(photos);
```

Comment:
Makes sense. So you just run the effect to getPhotos and then cacheImages with the new data in the state photo.

Reply:
Yeah, getPhotos() is triggered on initial render (like componentDidMount) and then once photos are set in the hook, it triggers the useEffect that calls cacheImages().